apis/v1alpha1/gateway_types.go

```diff
@@ -281,6 +283,20 @@ type Listener struct {
 	// are compatible with the application protocol specified in
 	// the Protocol field.
 	//
+	// Multiple Routes may match a request to this Listener. If that happens,
+	// precedence MUST be given in order of the following criteria:
+	// * The match with the longest combination of hostname and path.
+	// * For conflicts within a Route, the match appearing first.
```

Comment:
Is this technically feasible to do in most implementations?

Reply:
Good point, it's not. It seems to be a select few where order is meaningful. I've removed this.